src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala

```diff
@@ -310,4 +321,109 @@ final case class AgreementsApiServiceImpl(
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerFile: ToEntityMarshaller[File]
+  ): Route = {
+    logger.info(s"Retrieving consumer document $documentId from agreement $agreementId")
+
+    val result: Future[HttpEntity.Strict] =
+      for {
+        agreementUUID <- agreementId.toFutureUUID
+        documentUUID  <- documentId.toFutureUUID
+        document      <- agreementProcessService.getConsumerDocument(agreementUUID, documentUUID)
+        contentType   <- getMediaType(document.contentType, agreementId, documentId)
+        byteStream    <- fileManager.get(ApplicationConfiguration.storageContainer)(document.path)
+      } yield HttpEntity(contentType, byteStream.toByteArray())
+
+    onComplete(result) {
+      handleError(s"Error downloading contract fro agreement $agreementId") orElse { case Success(contract) =>
+        complete(contract)
+      }
+    }
+  }
+
+  private def getMediaType(contentType: String, agreementId: String, documentId: String): Future[ContentType] =
+    ContentType
+      .parse(contentType)
+      .leftMap(errors => InvalidContentType(contentType, agreementId, documentId, errors))
+      .toFuture
+
+  override def getAgreementContract(agreementId: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerFile: ToEntityMarshaller[File]
+  ): Route = {
+    logger.info(s"Retrieving contract for agreement $agreementId")
+
+    val result: Future[HttpEntity.Strict] =
+      for {
+        uuid       <- agreementId.toFutureUUID
+        agreement  <- agreementProcessService.getAgreementById(uuid)
+        contract   <- agreement.contract.toFuture(ContractNotFound(agreementId))
+        byteStream <- fileManager.get(ApplicationConfiguration.storageContainer)(contract.path)
+      } yield HttpEntity(ContentType(MediaTypes.`application/pdf`), byteStream.toByteArray())
+
+    onComplete(result) {
+      handleError(s"Error downloading contract fro agreement $agreementId") orElse { case Success(contract) =>
+        complete(contract)
+      }
+    }
+  }
+
+  override def removeAgreementConsumerDocument(agreementId: String, documentId: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = {
+    logger.info(s"Deleting consumer document $documentId for agreement $agreementId")
+
+    val result: Future[HttpEntity.Strict] =
+      for {
+        uuid       <- agreementId.toFutureUUID
+        agreement  <- agreementProcessService.getAgreementById(uuid)
+        contract   <- agreement.contract.toFuture(ContractNotFound(agreementId))
+        byteStream <- fileManager.get(ApplicationConfiguration.storageContainer)(contract.path)
+      } yield HttpEntity(ContentType(MediaTypes.`application/pdf`), byteStream.toByteArray())
```

Comment:
this endpoint should delete the document but it is actually a retrieve

Reply:
+1